### DISCUSSION TITLE
I really have no idea how to use this software. Any step-by-step instructions?

### DISCUSSION THREAD

**Discussion by weishanlee** [TheAxeDude/ZiniGo#21] [Q&A]
I think them very useful codes but I really do not know how to use them even if I read the instructions. Can anyone provide a step-by-step demo?
Thank you very much
wei shan

**Comment by TheAxeDude** [comment]
Hi there, I'll try set up a step-by-step guide this weekend :)

**Reply by MeNotSoSmart** [reply]
Hello. Did you happen to write up a step-by-step guide? I kind of got it working, but it does not download the items in my Zinio downloaded magazines, I am not sure why it is downloading what it is downloading, it seems to be downloading random items. I am sure I have done something wrong, just not sure what I have done. Thank you for your help and for making this program.

Edit: ZiniGo seems to be downloading all magazines related to my account including archived.

**Reply by royhuang1** [reply]
Any update on the write-up? Thanks!

**Reply by digitalfreaknyc** [reply]
Would love an update as well!

**Reply by DaveH52** [reply]
I'm stumped also! I have tried using several different incantations of 
./zinigo -u=Username -p=Password [-c=/path/to/chrome/executable]
but all I receive is 
GettingLogin
GotLogin
Fetching Library
Fetching page:1
**{"status":false,"error":{"message":"Invalid Json Web Token in AsyncMiddleware","internal_code":"INVALID_TOKEN"}}**
Terminating the application...

**Reply by motoyugota** [reply]
Here's the simplest set of instructions to get it running (in Windows):

1. Put executable in a folder of your choice
2. Create a file in the folder named "config.json"
3. Open config.json in a text editor and enter the following (Change the chromepath as necessary):

```
{
  "username":"[YOUR ZINIO USERNAME]",
  "password":"[YOUR ZINIO PASSWORD]",
  "chromepath":"C:\Program Files\Google\Chrome\Application\chrome.exe"
}
```

4. Open a command prompt
5. Type in `cd "[PATH TO YOUR FOLDER]"` and hit Enter
6. Type in `Zinigo_Windows_x64` and hit Enter

The files should generate in an "issue" subfolder

**Reply by dtschopp** [reply]
Thanks for the instructions. I get the following error:

Starting the application...
panic: runtime error: index out of range [2] with length 0

Is there something I could do about it?

**Reply by gmlay78** [reply]
I get the following:

GotLogin
Fetching Library
Fetching page:1
{"status":false,"error":{"message":"Invalid token","internal_code":"APPERR5555"}}
Terminating the application...

**Reply by wiildstar** [reply]
> I get the following:
> 
> GotLogin Fetching Library Fetching page:1 {"status":false,"error":{"message":"Invalid token","internal_code":"APPERR5555"}} Terminating the application...

@TheAxeDude 
I get the same error message. Any solution/work-around?

**Reply by wiildstar** [reply]
@TheAxeDude 

I think the "invalid token" is caused when you are still logged into the desktop reader app. When I log out of the app, and then rerun, I now get this error:

==================

cookie name:zwrrtcookie valueeyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhIjoiNTAyOTc0YTk0MmVkY2Q3MDY5Y2M0ZmExNDkyNmY5ZDA6ZWQ2ZjczNzAwYWViMjJlYzU0OGMxMTZjMzY2NWE5MzI1MDg0YzVmN2IyYmQxZjEzNDc1ZmEwZDNlNzFmNWFlZjFlNmY5MmI3M2M0YWI5ZDUyM2NjZmYyNWRmNTY0MzczMzliNGFiMjQ2ZDA5YjgwZGUwNmRkZGQ0NzFlZmU0NTdhNzUwYmMwYmI2YWY0OTNjOWJmYjRjOGM1MmRmMDMyNzVjYzg3NTNkMGRlOGEyNzU4NjljNDE1NzZmMWUxODNlNTdiNzc0NjRhMmE5ZGEyMTUwN2JlNDk4ZWE5ODBmODM5ODQwOTAxNmY1MGZlZmE0MjgwNzc5MTFmNWFkZjc4ODY2ZDViN2Y3MmM3MTZiNjI0MWFkZjg5YjRiNDlkZDBiODc2ZmQxYWVkNjM3MzEwN2VkMjhkNTU1MzI5NzE4ZGY0N2Y3MTdhNWYxNDg1YmI1MDc5MDAxZjMzMmUyMGIzMDk4YTg3MGIzMzA2MzhkY2MwYWJmZjY4M2FhOTI5ODNlMWJkZGU3YzUxM2NjMjRmNmI3M2IwNDRmMmQ1NTI3NTQxMTljMmY0ODU1NWJjYjNiNDQwOTQwMTQ1N2Y4NzA4MjQzZWJhZjM4ZDNmNzEzYjhkMjVkY2UwODU0ZTNmNzc2OWM3NDcwNjdlYzY2MDRlN2FjYzI0OTg1ZjljZTE0ZGQyZDQ3MTI2ZDQ4NzdjYTFhMDIyY2VhMDBkZjYxMWQ5NDNjYjA2ZWY3NmM4ZWMzMmZlY2E0YzRjOTM1YmViMDA1NjcxMmEzZWRhOTE4MWU2ODZhYjRiZmNjZDBjZmFiYmJlNDk1ODgxNzc3NTc0ZDM5ZmYyMWNlNjQwMWZhYTU1Y2M2ZWI0YWYzMDY5MmIzYjM5MGZlZTY4MjAzNTQ1MTc4MDkzM2Q3MjNiNzg0OTI3OWU2YzFiZjBiOWIxZDQ2YWMxZjI4MjMzZjQwNTMxODEwMGYwNDI4YzI0M2JkNDFiOTE2Yzk3MmZjYmZhMzhjMjVlMjBjYTYzZGVlOGQ2Mjc2OWZiYjkyNmY3YzM3OGY3NjBmYjI5ZTdkNTIxMzExYzNmOTc5ZGM1ZWZmNDc4ZWZlMTY0MzU1OTIwMmJmZmE2MGVmYjQ1MTA4OWU0NzQyYWYwYjA3ZGZjOWUzMWNmNDEwODE4NzFhYTU0Y2FiNDQ2ZTEyZDNjZWZlYzU3ZTczMTAxZjRmOGMyZDA4ZjQ5OTFiY2NhZGNlMGNiOTdjMzRjNiIsImlhdCI6MTcxODkxMjg2NSwiZXhwIjoxNzIxNTA0ODY1fQ.icl9vT4B37ItVTz12I2egIlnVfRTe5tqQLkx1vuUn3c
cookie name:zwrtcookie valueeyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhIjoiOTNkMDgyN2I5YmI4ODUxYTA1MzJhMTc0YjBlMzI4Y2U6Y2VkZDg2OGE3YzE1OWFhOWQ0OTgyOWM0YTg3MzE3Mzg5YWYwMDk0ZjYzMmQxOTk1YzM2OTQ1YjM4YzcwYmQ0MzQwYzY5YTE1MzRkMjkwMjIxMzJkMmEyZjNiYjczYmM2N2IwOWViYzdiNmIzZTVjY2NiOGY5NmEwOWFmY2U4NzU4OTliZmEwNmExZDhiM2U4Njk2NmU5YmM3YjBiNjliNzBlYTNlZTBkNWE3NDU3YjY2YzY1ODZiMmQzMzU3ODk5NDU3YWEzYTNhZjljYzRlYWY1Mjg4NmRjNjEzZTg5YWZiNDgyNWFkNDQwMzY3YjgxYzVhNDI5ZmVmNDAwY2ViMmJjNzM4ZDUwZDI1MjQzOGVlZDg5MDUwYTc3MzUyOTg1YzU5ZDE4ZDkyNjY5MmNjNjNiMjk3MWRhMzRlNTI3NzU4ZjY5MzgzNmQ0ZTA1MWFlZjBlYjcxZjE1YzI4YjhhYjRiNDhkOTRkN2JjOGQ3NzMxMzFlNGMyNTU4N2I4OGY5YmVlNTg4NGIwMDdlZGZiY2U4MzZhMWY4YTg0YTgyNDY0NzVkMTFjYTgwMWJhZjMzZTZkOTczZmVkNTYyYjBjMjI0NWUyMDgwZTJkNTIwMDllMjdjZTlhOWI0ODNmODJmYjQzMTAyM2ZjZDYyMzgwNWMwMjJhNmYzMThiZDMzYzcwZjk2Nzc4MmIxNmU5Y2ViYmNiZmRlYjJhNjc0NWM1ZTZhNDJhYjUyY2IwZjFhMDJjNTUzOWFlMDllZTIxODI4YzYxYjNmODk5MmRiM2Y1NWMxOGNiMTc5OGM3MWM4YTc2OGQ4MmJhOGZhOTk0MmU3ZjM3OTFmOTJhYThmODFlMWQ4NjE5YWUyNmFkMDhlZGZmYWE4MjdjYzVlMGJlNjhhMWUxN2FiODlhN2I3YzQ4ODFlNDZjOWZhM2FkZjBmNmRiOGIxMDMzOGRhZDNkNGVkOWE2MTU5MzEwODMwNGE1MWE5OTdhMmZjMDBmMGU0NjkyMmY3NzY1ZjBjN2VmMGI3NzljZjkwODAzZTI5ZjRhNDBlODNjZWY0NzQxZTdkMTVhY2M4YzI3YWM0ODA1MjlkNWM2NWFmM2UwOTQ4YTJmMmQzZTBkZmRiNjdjYWQ4NzRmODVkZDIzMGFlYjM2OGU0Yzc0NzM3ZjQ1YWQ2ZjBjOTVlOTczZGEyYTVlMjQyNTMyZTQwYmZmOWE4NGFhOTA0ZDc2MWZhYmZjMGI1OWI4YSIsImlhdCI6MTcxODkxMjg2NSwiZXhwIjoxNzE4OTIwMDY1fQ.wWTLD8Phs8sz2fPXI1mOgQtRzwZd26xVabxS9CkvTYY
cookie name:_hccookie value174066c09e79fe55
cookie name:zogincookie valuetrue
GotLogin
Fetching Library
Fetching page:1
{"status":false,"error":{"message":"Route not found in ApiRouter","internal_code":"APPERR13131313"}}
Terminating the application...

**Reply by marcogiaguaro** [reply]
@TheAxeDude my compliments for the application, i really need that.
But: 

F:\Zinigo>Zinigo_Windows_x64
Config file loaded
Username taken from config file
password taken from config file
chromepath taken from config file
No fingerprint found in text file, generating and writing
GettingLogin
{"email":"[xxxxxxxxxxxxxxxx]","password":"[xxxxxxxxxxxxxxxxx]","device":{"name":"Windows Chrome","fingerprint":"abcd123","device_type":"Desktop","client_type":"Web"},"newsstand":{"currency":"ZAR","id":134,"country":"ZA","name":"South Africa","cc":"za","localeCode":"en_ZA","userLang":"en_ZA","userCountry":"ZA","userCurrency":"ZAR","requiresCookies":true,"requiresExplicitConsent":true,"requiresAdultConfirmation":true,"adWords":{"id":1,"label":""},"isDefaultNewsstand":false}}
GotLogin
Fetching Library
Fetching page:1
{"status":false,"error":{"message":"Invalid token","internal_code":"APPERR5555"}}
Terminating the application..


> Here's the simplest set of instructions to get it running (in Windows):
> 
> 1. Put executable in a folder of your choice
> 2. Create a file in the folder named "config.json"
> 3. Open config.json in a text editor and enter the following (Change the chromepath as necessary):
> 
> ```
> {
>   "username":"[YOUR ZINIO USERNAME]",
>   "password":"[YOUR ZINIO PASSWORD]",
>   "chromepath":"C:\Program Files\Google\Chrome\Application\chrome.exe"
> }
> ```
> 
> 4. Open a command prompt
> 5. Type in `cd "[PATH TO YOUR FOLDER]"` and hit Enter
> 6. Type in `Zinigo_Windows_x64` and hit Enter
> 
> The files should generate in an "issue" subfolder

**Reply by marcogiaguaro** [reply]
@TheAxeDude it changed (i closed the desktop app):

Config file loaded
Username taken from config file
password taken from config file
chromepath taken from config file
Fingerprint taken from config file
GettingLogin
{"email":"[xxxxxxxxxxxxxxxxx]","password":"[xxxxxxxxxx]","device":{"name":"Windows Chrome","fingerprint":"vughqn20IlKFf9O","device_type":"Desktop","client_type":"Web"},"newsstand":{"currency":"ZAR","id":134,"country":"ZA","name":"South Africa","cc":"za","localeCode":"en_ZA","userLang":"en_ZA","userCountry":"ZA","userCurrency":"ZAR","requiresCookies":true,"requiresExplicitConsent":true,"requiresAdultConfirmation":true,"adWords":{"id":1,"label":""},"isDefaultNewsstand":false}}
GotLogin
Fetching Library
Fetching page:1
{"status":false,"error":{"message":"Invalid token","internal_code":"APPERR5555"}}
Terminating the application...

**Comment by jhidalgo27** [comment]
Hello, I have the same problem.

==========
jZjMGYyYSIsImlhdCI6MTcyMDk3NDM3MCwiZXhwIjoxNzIwOTgxNTcwfQ.1oWy3r0NCi01I2UEKzPKIY3q4mEhuHw7TjH-03YteOc
cookie name:_hccookie valuee827d15a756fa44c
cookie name:zogincookie valuetrue
GotLogin
Fetching Library
Fetching page:1
{"status":false,"error":{"message":"Route not found in ApiRouter","internal_code":"APPERR13131313"}}

**Comment by muttmutt** [comment]
Same problem:

`cookie name:_hccookie valueea91574fa4d88a00
cookie name:zogincookie valuetrue
GotLogin
Fetching Library
Fetching page:1
{"status":false,"error":{"message":"Route not found in ApiRouter","internal_code":"APPERR13131313"}}
Terminating the application...`

**Comment by muttmutt** [comment]
FWIW I also tested on Ubuntu 22.04 Linux with the same result.

**Comment by marcogiaguaro** [comment]
@TheAxeDude @motoyugota @dtschopp @DaveH52 @weishanlee  anyone have news about it?
